server/src/test/position.test.ts

```diff
@@ -0,0 +1,244 @@
+                ),
+                expectedResult: false,
+            },
+        ];
+
+        cases.forEach((x) => {
+            const expectedResultString = x.expectedResult
+                ? "should"
+                : "shouln't";
+            it(`position ${positionToString(
+                x.position
+            )} ${expectedResultString} fall into the range ${rangeToString(
+                x.range
+            )}`, () => {
+                expect(positionInRange(x.position, x.range)).to.equal(
+                    x.expectedResult
+                );
+            });
+        });
+    });
+
+    describe("forEachAvailableObjectForLink", () => {
+        class OmtDocumentInformationProviderStub {
+            private tsConfigFiles: string[] = [];
+            private workspaceLookup = null as unknown;
+
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            provideDocumentInformation(..._: unknown[]){
+                return {documentLinks: [], calledObjects: [], definedObjects: [], availableImports: []};
+            }
+
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            provideAvailableObjectsFromDocument(..._: unknown[]){
+                return {definedObjects: [], availableImports: []};
+            }
+
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            contextPaths(..._: unknown[]) {
+                return new Map<string, string>();
+            }
+
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            getOmtDocumentInformation(..._: unknown[]) { 
+                return {documentLinks: [], calledObjects: [], definedObjects: [], availableImports: []};
+            }
+        }
+        const omtDocumentInformationProvider = (new OmtDocumentInformationProviderStub() as unknown) as OmtDocumentInformationProvider;
+
+        const linkUrlString = "file/file1.txt";
+        
+        const link = {
+            name: "HelloWorld",
+            range: Range.create(
+                { line: 0, character: 10 },
+                { line: 0, character: 20 }
+            ),
+            parameters: [],
+        };
+        const callbackfn = (
+            linkUrl: string,
+            omtLocalObject: OmtLocalObject
+        ) => ({ linkUrl, omtLocalObject });
+        
+        it("should return no objects when empty", () => {
+            // ARRANGE
+            const availableObjects = {
+                definedObjects: [],
+                availableImports: [],
+            };
+
+            // ACT
+            const result = forEachAvailableObjectForLink(
+                omtDocumentInformationProvider,
+                linkUrlString,
+                availableObjects,
+                link,
+                callbackfn
+            );
+
+            // ASSERT
+            expect(result.length).to.equal(0);
+        });
+
+        it("should return one objects when defined", () => {
```

Comment:
not very important, but still
```suggestion
        it("should return one object when defined", () => {
```